packages/client/src/components/Post.js

```diff
@@ -0,0 +1,132 @@
+import { useState, useEffect } from 'react';
+import mockPost from './mockuser';
+import styled from 'styled-components';
+import { Eye, EyeSlash } from 'phosphor-react'
+
+const Button = styled.button`
+background-color: black;
+border: none;
+color: white;
+width: 12rem;
+padding: 1rem 0;
+&:hover {
+    background-color: #e5e5e5;
+    color: black;
+}
+font-size: 1.875rem;
+`;
+
+const ButtonDiv = styled.div`
+display: flex;
+justify-content: space-evenly;
+margin-top: 1.5rem;
+`
+const PostHeading = styled.div`
+text-align: left;
+background-color: #e5e5e5;
+padding: 1rem 3rem;
+display: flex;
+justify-content: space-between;
+align-items: center;
+z-index: 1000;
+h2 {
+    font-size: 3rem;    
+}
+`;
+
+const Body = styled.section`
+    padding: 4em;
+    color: black;
+    width: 100%;
+`;
+
+const PostContainer = styled.div`
+width: 100%;
+height: 100vh;
+padding: 1.5rem;
+display: flex;
+flex-flow: column nowrap;
+justify-content: center;
+align-content: center;
+position: relative;
+`;
+
+let isAdmin = true;
+
+const Post = ({children}) => {
+    const [title, setTitle] = useState("")
+    const [body, setBody] = useState("")
+    const [isCollapsed, setIsCollapsed] = useState(true)
+
+    const styles = {
+        show:{
+            transform: 'scaleY(1)',
+            transformOrigin: 'top',
+            position: 'static',
+            overflow: 'hidden',
+            webkitTransition: `transform 200ms`,
+            transition:`transform 200ms`
+        },
+        hide:{
+            transform: 'scaleY(0)',
+            transformOrigin: 'top',
+            webkitTransition: `transform 300ms ease-in-out 200ms`,
+            transition:`transform 300ms ease-in-out 200ms`
+        },
+        showChildren: {
+            opacity: 1,
+            transition:`opacity 150ms ease-in-out 150ms`,
+        },
+        hideChildren: {
+            opacity: 0,
+            transition:`opacity 100ms ease-in-out`            
+        }
+    }
```

Comment:
We can move this to styled-components. Not something that _really_ matters. Example:
```js
const CollapsableDiv = styled.div`
  {({ hidden }) => {
    if (hidden) {
      return `transform: scaleY(0);
      transform-origin: top;`;
    }
  }}
`;
```

Reply:
Even if you decide this is better let's move the styles outside of the component else these styles will be recreated  every rerender of the post